app/lib/pre_assembly/batch.rb

```diff
@@ -112,17 +112,10 @@ def process_digital_objects
         num_no_file_warnings += 1 if dobj.object_files.empty?
         progress = { dobj: dobj }
         file_attributes_supplied = batch_context.all_files_public? || dobj.dark?
-        begin
-          # Try to pre_assemble the digital object.
-          load_checksums(dobj)
-          status = dobj.pre_assemble(file_attributes_supplied)
-          # Indicate that we finished.
-          progress[:pre_assem_finished] = true
-          log "Completed #{dobj.druid}"
-        ensure
-          # Log the outcome no matter what.
-          File.open(progress_log_file, 'a') { |f| f.puts log_progress_info(progress, status || incomplete_status).to_yaml }
-        end
```

Comment:
we will now do a rescue from within the `DigitalObject#pre_assemble` and return status directly

Reply:
the `ensure` clauses seem like a good thing ... why are we jettisoning them?

Reply:
I can put it back, but I wasn't sure what it was actually doing...since there is no associated rescue block above it?